core/src/main/java/amino/run/kernel/common/ServerInfo.java

```diff
@@ -17,8 +17,11 @@
 
     private Map<String, String> labels = new HashMap<String, String>();
 
+    public int processorCount; // Available processor count
+
     public ServerInfo(InetSocketAddress addr) {
         this.host = addr;
+        this.processorCount = Runtime.getRuntime().availableProcessors();
```

Comment:
This will only work correctly when this constructor is called on the machine to which the server info refers.  Surely processorCount should be a parameter to the constructor, similar to addr?